### PR TITLE
Fix -pkeyopt handling in apps/pkeyutl -rawin

### LINKED ISSUE #14169: -pkeyopt argument not recognized by pkeyutl -sign -rawin
When I try using `-pkeyopt` with `openssl pkeyutl -sign -rawin` I get an error:
```
$ echo "ABC" | openssl pkeyutl -sign -inkey test/testrsa.pem -rawin -pkeyopt rsa_padding_mode:pss
openssl pkeyutl: Can't set parameter "rsa_padding_mode:pss":
C061A8F8077F0000:error:03000093:digital envelope routines:evp_pkey_ctx_ctrl_str_int:command not supported::0:
```
This is due to a wrong order of initialization, because the CTX has to be initialized before the `evp_pkey_ctx_ctrl_str_int` is called. I already have a patch for this: #13828. (The patch is ready, but I forgot to associate it with an issue.)

### PR DESCRIPTION
**Pull request by gotthardp**
Fixes #14169. The EVP_DigestSignInit and EVP_DigestVerifyInit have to be initialized before the EVP_PKEY_CTX_ctrl_str is invoked. The statement about delayed initialization (introduced by https://github.com/openssl/openssl/commit/ee633ace7397ebc9829b845e7e7bbd8283b455f6) feels wrong. When the ctx not initialized properly the ctrl command used to set `-pkeyopt` fails.
This PR fixes handling of the `-pkeyopt` argument when `-rawin` is also defined by moving EVP_DigestSignInit and EVP_DigestVerifyInit before EVP_PKEY_CTX_ctrl_str.

##### Checklist

- [x] tests are added or updated

### CODE REVIEW

**Comment by beldmit** [comment]
Would you mind to add a test?

**Comment by gotthardp** [comment]
Yeah, sure! It's there now.

**Comment by gotthardp** [comment]
@mattcaswell, would you please review this patch? It modifies the code you contributed, so I wonder if my assumptions are correct.

File: apps/pkeyutl.c
Comment (gotthardp) [reply apps/pkeyutl.c]: Yeah, you're right. I was too paranoid. Thanks!

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
ping @openssl/committers for the second review

**Review by beldmit — APPROVED** [review]
LGTM

**Comment by t8m** [comment]
Merged to master. Thank you for the contribution!

**Comment by gotthardp** [comment]
Thank you very much for your help with this!

**Pull request closed without merging**
